traitsui/testing/tester/qt4/implementation/list_editor.py

```diff
@@ -8,14 +8,90 @@
 #
 #  Thanks for using Enthought open source!
 #
-from traitsui.testing.tester import locator
+from traitsui.testing.tester import command, locator
 from traitsui.testing.tester.registry_helper import register_nested_ui_solvers
-
+from traitsui.testing.tester.qt4 import helpers
 from traitsui.qt4.list_editor import (
     CustomEditor,
+    NotebookEditor,
 )
 
 
+class _IndexedNotebookEditor:
+    """ Wrapper for a ListEditor (Notebook) with an index.
+    """
+
+    def __init__(self, target, index):
+        """
+        Parameters
+        ----------
+        target : NotebookEditor
+            The Notebook List Editor
+        index : int
+            The index of interest.
+        """
+        self.target = target
+        self.index = index
+
+    @classmethod
+    def from_location(cls, wrapper, location):
+        """ Helper method to create an instance of _IndexedNotebookEditor
+        but checking if the index is valid before doing so.
+
+        Parameters
+        ----------
+        wrapper : UIWrapper
+            the UIWrapper wrapping the Notebook List Editor
+        location : locator.Index
+            the locator.Index object containing the index
+        """
+        # Raise IndexError early
+        wrapper.target._uis[location.index]
+        return cls(
+            target=wrapper.target,
+            index=location.index,
+        )
+
+    @classmethod
+    def register(cls, registry):
+        """ Class method to register interactions on a _IndexedNotebookEditor
+        for the given registry.
+
+        If there are any conflicts, an error will occur.
+
+        Parameters
+        ----------
+        registry : TargetRegistry
+            The registry being registered to.
+        """
+        registry.register_solver(
+            target_class=NotebookEditor,
+            locator_class=locator.Index,
+            solver=cls.from_location,
+        )
+
+        register_nested_ui_solvers(
+            registry=registry,
+            target_class=cls,
+            nested_ui_getter=lambda target: target._get_nested_ui()
+        )
+
+        registry.register_handler(
+            target_class=cls,
+            interaction_class=command.MouseClick,
+            handler=lambda wrapper, _: helpers.mouse_click_tab_index(
+                tab_widget=wrapper.target.target.control,
```

Comment:
this `wrapper.target.target.control` is ugly, but it will eventually be fixed once the refactor mentioned in the previous comment is done.  For now it does what we want it to.